### PR TITLE
[Bugfix] Fix DeepSeek v4 topk numerical issue for unaligned max-model-len

### PR DESCRIPTION
**Pull request by wzhao18**
## Purpose:
It is observed that DeepSeek v4 has numerical issues when passing certain values of `--max-model-len` (e.g., 900000). For instance, the following figure shows the distribution of the number of completion tokens with 100 responses sampled for a selected question from GPQA. It can be seen that the number of generated tokens has a noticeable distribution shift when `--max-model-len 900000` is passed, compared to that not passed. 

rank_completion_tokens

Investigation shows that the root cause is that the indexer persistent topk kernel uses `const int64_t stride = logits.size(1);`. However, the logits produced from [deepgemm's](https://github.com/deepseek-ai/DeepGEMM/blob/891d57b4db1071624b5c8fa0d1e51cb317fa709f/csrc/apis/attention.hpp#L361) `fp8_fp4_paged_mqa_logits` pads the `max_model_len` dimension to align with 256 and later slices the view to `max_model_len`. Thus, when `max_model_len` is not naturally aligned, the logits become incontiguous, thus the stride calculation is incorrect. This PR fixes this bug.

To reproduce the experiment:
```
VLLM_ENGINE_READY_TIMEOUT_S=3600 \
vllm serve deepseek-ai/DeepSeek-V4-Pro \
--kv-cache-dtype fp8 \
--trust-remote-code \
--block-size 256 \
--no-enable-prefix-caching \
--compilation-config '{"cudagraph_mode":"FULL_AND_PIECEWISE","custom_ops":["all"]}' \
--attention_config.use_fp4_indexer_cache True \
--tokenizer-mode deepseek_v4 \
--tool-call-parser deepseek_v4 \
--enable-auto-tool-choice \
--reasoning-parser deepseek_v4 \
--max-cudagraph-capture-size 2048 \
--max-num-batched-tokens 2048 \
--max-model-len 900000 \
--no-enable-flashinfer-autotune \
--tensor-parallel-size 8
```

Set and unset `--max-model-len 900000` to see the difference.

Prompt:
```
QUESTION = (
    "You have prepared a di-substituted 6-membered aromatic ring compound "
    "containing 8 carbon atoms in total. Peaks in the FTIR spectrum of this "
    "compound indicate the presence of a carbonyl group as well as an "
    "aromatic-halogen bond. Identify the correct 1H NMR data for this compound. \n"
    "\n"
    "A) 1H NMR: chemical reference (ppm): 6.9 (1H, s), 4.8 (2H, d), 4.6 (2H, d), 1.3 (2H, s)\n"
    "B) 1H NMR: chemical reference (ppm): 9.9 (1H, s), 7.8 (2H, d), 7.6 (2H, d), 3.7 (2H, s)\n"
    "C) 1H NMR: chemical reference (ppm): 4.8 (2H, d), 4.6 (2H, d), 1.3 (3H, s)\n"
    "D) 1H NMR: chemical reference (ppm): 7.8 (2H, d), 7.6 (2H, d), 2.3 (3H, s)"
)
```

## Test Plan
- Test the num token distribution is normal
- Test e2e eval accuracy

## Test Result
The figure shows that after applying the patch, the num token distribution becomes normal.

In addition, e2e GPQA results:
```
main TP without max-model-len:
nemo-run_1/0 ----------------------------------------- gpqa ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 198         | 11754      | 3361        | 89.39%           | 0.00%    

main DEP without max-model-len:
nemo-run_1/0 ----------------------------------------- gpqa ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 198         | 12232      | 4058        | 88.88%           | 0.00%    

main TP with --max-model-len 900000:
nemo-run_1/0 ----------------------------------------- gpqa ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 198         | 25955      | 12300       | 81.82%           | 5.05%    

main DEP with --max-model-len 900000:
nemo-run_1/0 ----------------------------------------- gpqa ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 198         | 17770      | 7457        | 89.39%           | 0.00%    

branch TP with --max-model-len 900000:
nemo-run_1/0 ----------------------------------------- gpqa ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 198         | 12573      | 3514        | 88.89%           | 0.00% 

branch DEP with --max-model-len 900000:
nemo-run_1/0 ----------------------------------------- gpqa ----------------------------------------
nemo-run_1/0 evaluation_mode | num_entries | avg_tokens | gen_seconds | symbolic_correct | no_answer
nemo-run_1/0 pass@1          | 198         | 12381      | 4197        | 87.88%           | 0.00%    
 
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by ywang96** [comment]
FYI @LopezCastroRoberto @zyongye

**Review by zyongye — APPROVED** [review]
